### PR TITLE
[SPARK-29150][CORE] Update RDD API for Stage level scheduling to be public

### PR DESCRIPTION
**Pull request by tgravescs**
### What changes were proposed in this pull request?

This PR is to make the RDD api for stage level scheduling public.  All the other jiras for functionality are in so now we can make it public for people to use and tag it properly.

### Why are the changes needed?

to make it usable. 

### Does this PR introduce _any_ user-facing change?

Yes the api is now public

### How was this patch tested?

Unit tests and manually tested.

### CODE REVIEW

**Comment by tgravescs** [comment]
test this please

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thanks, @tgravescs .

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: core/src/main/scala/org/apache/spark/rdd/RDD.scala [hunk 1]
```diff
@@ -1780,10 +1780,9 @@ abstract class RDD[T: ClassTag](
    * It will result in new executors with the resources specified being acquired to
    * calculate the RDD.
    */
-  // PRIVATE for now, added for testing purposes, will be made public with SPARK-29150
   @Experimental
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/rdd/RDD.scala:1783]: Thanks for adding this issue. I was about to suggest to keep it same `Evolving` but I suspect there's a reason for it :-). LGTM

**Review by HyukjinKwon — APPROVED** [review]
Thanks @tgravescs. LGTM

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
I am going to increase timeout for `test_streaming_algorithms` for now...

**Comment by HyukjinKwon** [comment]
Merged to master.

File: core/src/main/scala/org/apache/spark/rdd/RDD.scala (hunk 1 above)
Comment (tgravescs) [reply core/src/main/scala/org/apache/spark/rdd/RDD.scala:1783]: this one I just missed originally

**Comment by tgravescs** [comment]
thanks for the reviews

**Pull request closed without merging**